inst/string.m

```diff
@@ -148,23 +149,30 @@
           error ('string: cell inputs must be cellstr');
         endif
         this.strs = in;
-        this.tfMissing = false (size (this.strs));
+        tfM = false (size (this.strs));
+        tfM(cellfun (@(x) isempty(x), this.strs)) = true;
```

Comment:
I'm not sure `tfM(cellfun (@(x) isempty(x), this.strs)) = true;` is right for the case of cellstr `x` inputs. That `isempty(x)` will be true for 0-character strings like `''`. But a blank/empty string isn't the same as a missing or NaN, and those don't convert to `string(missing)` values; they convert to `""` empty string elements (a non-missing `string` that represents a 0-character-long character sequence). Kinda like how for doubles, 0.0 and NaN aren't the same thing. 

AFAIK, in Matlab, cellstrs don't strictly have an equivalent of a missing value. In not-really-a-cellstr cell arrays, I've seen a numeric `[]` cell value used to represent "really missing/null as opposed to the empty string. And in some contexts, like SQL data retrievals, NULL values are mapped to certain strings like `''` or `'NULL'` and then the client deals with those sentinel values. But that's context-specific, and I don't think it should happen in the string type itself.

Reply:
I did this so that we can catch the default missing values. The way it isimplemented, `this.strs` contains an empty syting `""` , but `this.tfMissing` becomes `true`. This way, when we call `table` to construct an empty table of given size, it shows that it is empty for `string` variabletype

Reply:
Hmm. I'm not sure I follow.

Maybe this is a terminology problem. As I understand it, in M-code, the term "empty" is usually applied to whole arrays, and means an array containing 0 elements, which is 0 long along at least one of its dimensions. Where `isempty(x)` and `numel(x) == 0` are true. And "missing" is a characteristic of an element (scalar value) of an array, like a NaN or NaT, which is null-like. An empty array does not contain any missing values (elements), because it doesn't contain any elements at all, either missing or nonmissing.

The term "empty string" applied specifically to string arrays is a little overloaded, because you can have a 0-long empty string array (`string([])`), and there's also a scalar string value `""` that represents a 0-long character sequence. That `""` would be represented by an empty `char` array in a char or cellstr representation, but as a `string` array it's not "empty" in the normal M-code sense of that. It's not missing, either; it represents a known character sequence and not a missing-value placeholder.

I've been sloppy and sometimes referring to `""` as an "empty string", because of analagy to the `char` representation and because I can't think of a better word for it. Not "blank", because AFAICT M-code uses "blank" to refer to space characters and other whitespace, so `".  "` would be a "blank" string that is 3 characters long, distinct from the 0-character `""`. Need a better term for that, for disambiguation.

AFAIK, chars and cellstrs do not have an equivalent of a `string(missing)` value; they are not NaN-able types.

So, when you say "construct an empty table of given size" here, do you mean a 0-row table with columns all containing 0 elements? Or do you mean a more-than-0-row table, where each column contains all missing values in its elements?

Reply:
> So, when you say "construct an empty table of given size" here, do you mean a 0-row table with columns all containing 0 elements? Or do you mean a more-than-0-row table, where each column contains all missing values in its elements?

According to MATLALB documentation, the latter applies.
````
sz = [4 3];
varTypes = {'double','datetime','string'};
T = table('Size',sz,'VariableTypes',varTypes)

T=4×3 table
    Var1    Var2      Var3   
    ____    ____    _________

     0      NaT     <missing>
     0      NaT     <missing>
     0      NaT     <missing>
     0      NaT     <missing>
````
The changes I 've made in `string` class were targeting the `table` functionality as shown above. In [this](https://www.mathworks.com/help/matlab/ref/table.html#mw_9cdbdca8-7493-4866-9e8d-c2ddb22e7a1f) documentation you can find the default values for `missing` vartypes. The table under the "varTypes — Data types of preallocated variables" header lists the default values and this is what I used as a guide in the changes in the `string` class.

Reply:
Ah. Yeah, I think that's the correct behavior for the `table` ctor with the `Size` argument. I just wouldn't call that an "empty" table; more like an "all-missing" or "fill" table.

I get the need to construct missing-valued `string` arrays for this purpose. But I don't think the cellstr-to-string variant of the `string` constructor is the place to handle that, because that's not compatible with how the Matlab `string` ctor works, and may cause problems in other areas where you need to convert char or cellstr arrays which contain `""` 0-long strings.

How about using `string.missing` or the `NaS` function, or the `tableOuterFillValue()` function, which is sort of a generalization of this behavior for when `table` needs missing-filled values for outer joins and similar operations?

AFAICT, in the Matlab API, there's no super-standardized way to produce these values, because the 0-arg ctors don't all return missings, there's no standard static method for producing missing-ish values, and not all datatypes support missingness/NaN-ness. The closest i've been able to get generically is: create or get a scalar value of that type and put it in variable `x`, then extend `x` with `x(3) = x`, then look at `x(2)` and that's your default "fill" value for that type: the type that gets filled in to array elements that are added ("vivified") but not assigned to during array expansion.